var/spack/repos/builtin/packages/sundials/package.py

```diff
@@ -27,138 +27,531 @@
+    variant('shared',  default=True,
+            description='Build shared libraries')
+    variant('static',  default=True,
+            description='Build static libraries')
+
+    # Fortran interface
+    variant('fcmix', default=False,
+            description='Enable Fortran interface')
+
+    # Examples
+    variant('examples-c', default=True,
+            description='Enable C examples')
+    variant('examples-cxx', default=False,
+            description='Enable C++ examples')
+    variant('examples-f77', default=True,
+            description='Enable Fortran 77 examples')
+    variant('examples-f90', default=False,
+            description='Enable Fortran 90 examples')
+    variant('examples-cuda', default=False,
+            description='Enable CUDA examples')
+    variant('examples-raja', default=False,
+            description='Enable RAJA examples')
+    variant('examples-install', default=True,
+            description='Install examples')
+
+    # Generic (std-c) math libraries (UNIX only)
+    variant('generic-math', default=True,
+            description='Use generic (std-c) math libraries on unix systems')
+
+    # xSDK options
+    variant('xsdk-defaults', default=False,
+            description='Use default xSDK configuration')
+
+    # ==========================================================================
+    # Conflicts
+    # ==========================================================================
+
+    # Options added after v2.6.2
+    conflicts('+hypre', when='@:2.6.2')
+    conflicts('+petsc', when='@:2.6.2')
+
+    # Options added after v2.7.0
+    conflicts('+cuda',          when='@:2.7.0')
+    conflicts('+raja',          when='@:2.7.0')
+    conflicts('+blas',          when='@:2.7.0')
+    conflicts('+indextype',     when='@:2.7.0')
+    conflicts('+xsdk-defaults', when='@:2.7.0')
+    conflicts('+examples-cuda', when='@:2.7.0')
+    conflicts('+examples-raja', when='@:2.7.0')
+
+    # External libraries incompatible with 64-bit indices
+    conflicts('+lapack', when='@3.0.0-beta-2: indextype=int64_t')
+    conflicts('+hypre',  when='+hypre@:2.6.1a indextype=int64_t')
+
+    # External libraries incompatible with single precision
+    conflicts('+klu',   when='precision=single')
+    conflicts('+hypre', when='+hypre@:2.12.0 precision=single')
+
+    # External libraries incompatible with extended (quad) precision
+    conflicts('+lapack',     when='precision=extended')
+    conflicts('+superlu-mt', when='precision=extended')
+    conflicts('+klu',        when='precision=extended')
+    conflicts('+hypre',      when='+hypre@:2.12.0 precision=extended')
+
+    # ==========================================================================
+    # Dependencies
+    # ==========================================================================
+
+    # Build dependencies
+    depends_on('cmake', type='build')
+
+    # MPI related dependencies
+    depends_on('mpi', when='+mpi')
+    depends_on('mpi', when='+hypre')
```

Comment:
can you build `~mpi+hypre`? if not, there should be a conflict

Reply:
You're right, this needs to be added as a conflict.